pkg/helm/release/manager.go

```diff
@@ -289,6 +293,17 @@ func generatePatch(existing, expected runtime.Object) ([]byte, error) {
 		return nil, nil
 	}
 
+	// We should add "remove" operations if the same path does also have
+	// a "add" operation
+	for _, rop := range removeOps {
+		for _, pop := range patchOps {
+			if rop.Path == pop.Path && pop.Operation == "add" {
+				patchOps = append(patchOps, rop)
+				break
+			}
+		}
+	}
+
```

Comment:
Also, see that the above is not covered by the tests. See: https://coveralls.io/builds/29873112/source?filename=pkg/helm/release/manager.go#L304

Reply:
added a new test case to cover this

Reply:
well no, i didn't, i cant replicate this issue with the new jsonpath library, should i remove this?

Reply:
Are you unable to replicate the issue locally or just in the tests? 
I mean, are you able to replicate any issue scenario locally where this code would solve the issue faced?

Reply:
Am I understanding correctly that it seems like the new library correctly chooses `replace` rather than adding both an `add` and `remove`?

Does that just mean this PR becomes just the dependency change to use the new library?

Reply:
And the added unit tests :pray: :slightly_smiling_face:

Reply:
Yes, i removed the additional for-loop. I think this issue is fixed in the new jsonpatch library